utils/mil_ros_tools/online_bagger/nodes/online_bagger.py

```diff
@@ -0,0 +1,427 @@
+#!/usr/bin/env python
+
+from __future__ import division
+import rospy
+import rosbag
+import rostopic
+import os
+import rospkg
+import time
+import math
+import resource
+from collections import deque
+import itertools
+import datetime
+from mil_ros_tools.srv import  BaggerCommands
+
+"""
+To call call service call the '/online_bagger/dump' service with the BaggerCommands.srv
+Amount and Unit. Doing this will generate a rosbag with the
+last x number of seconds saved
+
+For example:
+
+        bagging_server = rospy.ServiceProxy('/online_bagger/dump', BaggerCommands)
+        bag_status = bagging_server(x, 's')
+        bag_status = bagging_server(Amount, Unit)
+        amount = float32 (leave blank to dump entire bag)
+        unit = string (select units 's' or 'm')
+        """
+
+class OnlineBagger(object):
+
+    def __init__(self):
+
+        """
+        Make dictionary of dequeues.
+        Subscribe to set of topics defined by the yaml file in directory
+        Stream topics up to a given ram limit, dump oldest messages when limit is reached
+        Set up service to bag n seconds of data default to all of available data
+        """
+
+        self.i = 0  # successful subscriptions
+        self.n = 0  # number of iterations
+        self.streaming = True
+        self.dumping = False
+        self.get_params()
+        self.make_dicts()
+
+        self.bagging_service = rospy.Service('/online_bagger/dump', BaggerCommands,
+                                             self.check_action)
+
+        self.subscribe_loop()
+        rospy.loginfo('subscriber success: %s', self.subscriber_list)
+
+    def get_params(self):
+
+        """
+        Retrieve parameters from param server.
+        """
+
+        self.topics = rospy.get_param('/online_bagger/topics')
+        self.ram_limit = rospy.get_param('/online_bagger/ram_limit') * 1.073e9  # gigabytes to bytes
+        # self.stream_time = rospy.get_param('/online_bagger/stream_time')
+        self.dir = rospy.get_param('/online_bagger/bag_package_path')
+        self.bag_name = rospy.get_param('/online_bagger/bag_name')
+
+        if not self.bag_name:
+            self.bag_name = str(datetime.date.today()) + '-' + str(datetime.datetime.now().time())[0:8]
```

Comment:
Will this permanently set the name of the bag to the date when the program is started?